### DISCUSSION TITLE
How to remediate `NON_ESYNC_WINE_VERSION` on OpenSUSE TW?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [lutris/lutris#4979] [Q&A]
![image](https://github.com/lutris/lutris/assets/42837531/43037bfd-b692-492f-baa4-81fddc37b17b)

How might one go about remediating this? I am using cpe:/o:opensuse:tumbleweed:20230807. I installed using https://dl.flathub.org/repo/appstream/net.lutris.Lutris.flatpakref.

**Comment by danieljohnson2** [comment]
I think you need to use a different version of Wine. You can download one inside Lutris; there's a little icon in the sidebar, next to "Wine", to do this.

You can then configure your game to use it, and hopefully that will help.

**Reply by roke-julian-lockhart** [reply]
@danieljohnson2,

> You can download one inside Lutris

![image](https://github.com/lutris/lutris/assets/42837531/d323c05b-6013-45d0-b6a7-90dc5652f409)

Uh, this is a little overwhelming.

**Comment by danieljohnson2** [comment]
Grab the top one; that looks pretty good.

Lutris (I'm told) is meant to auto-download the top one whenever you install a game, if you don't have it, just so you'll have the latest. But that's bugged in 0.5.13; hopefully we'll do this properly in the next release.

**Reply by roke-julian-lockhart** [reply]
![image](https://github.com/lutris/lutris/assets/42837531/9cb63b69-0adf-4560-9ade-a15cf0e45ee4)

Doesn't appear to have worked, @danieljohnson2.

**Comment by danieljohnson2** [comment]
You need to select the new version of Wine in the game configuration too, I expect. Did you do that?

If you did so, did you the error change? There's a second thing that can fail here, something about ESYNC 'limits'. Maybe its that now?

**Reply by roke-julian-lockhart** [reply]
@danieljohnson2, https://github.com/lutris/lutris/discussions/4979#discussioncomment-6724437

> You need to select the new version of Wine in the game configuration too, I expect. Did you do that?

It was prefilled to the correct version, so I doubt that there's that anything I need to or am able to modify.

![image](https://github.com/lutris/lutris/assets/42837531/78b926bc-f08c-43a4-8a5e-64bbe24f20ca)

> did you the error change? There's a second thing that can fail here, something about ESYNC 'limits'

![image](https://github.com/lutris/lutris/assets/42837531/e1176fcc-9364-4920-8039-e1118ab483c8)

**Reply by danieljohnson2** [reply]
This doesn't seem right. I'll need to look into this.

**Comment by danieljohnson2** [comment]
So this is really weird. The check you are failing here actually looks at the *name* of the Wine version. Having "Proton" in the name should be enough to count.

The only thing I can think of is that Lutris lolwercases the name and looks for "proton"; maybe lowercasing it going wrong?

Try running this from the terminal:
```
python3 -c "print('Proton'.lower())"
```
And let me know what you get. It should be ``proton``.

This is a real shot in the dark though.

**Reply by strycore** [reply]
> maybe lowercasing it going wrong? 

I beg you to have better trust in your tools! :pray:

**Reply by danieljohnson2** [reply]
Trust but verify!

So far as I know there's nothing like the Turkish 'i' problem that could affect this word, but we are using lower() not casefold() so maaaaybe?

**Reply by roke-julian-lockhart** [reply]
@danieljohnson2, https://github.com/lutris/lutris/discussions/4979#discussioncomment-6734691

```pwsh
python3 -c "print('Proton'.lower())"
```

```log
PS /home/rokejulianlockhart> python3 -c "print('Proton'.lower())"
proton
```

```python3
print('Proton'.lower())
```

```log
PS /home/rokejulianlockhart> python3
Python 3.11.4 (main, Jun 28 2023, 19:51:46) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print('Proton'.lower())
proton
```

**Reply by danieljohnson2** [reply]
And rationality wins another round. Damn you rationality!

I do not understand what is going on here, but could you try the master branch? I have touched this area of the code, though I certainly never meant to fix a bug like this. I was fixing the double error message here.

Just download the master branch and run the 'bin/lutris' script in it- no need to install anything. Try your game, configured to use the Proton version as you have it, and see if that makes any difference.
